django/core/serializers/python.py

```diff
@@ -100,6 +100,21 @@ def Deserializer(object_list, *, using=DEFAULT_DB_ALIAS, ignorenonexistent=False
                 data[Model._meta.pk.attname] = Model._meta.pk.to_python(d.get('pk'))
             except Exception as e:
                 raise base.DeserializationError.WithData(e, d['model'], d.get('pk'), None)
+        else:
+            # If the model instance doesn't have a primary key and the model supports
+            # natural keys, try to retrieve it from the database.
+            default_manager = Model._meta.default_manager
+
+            if hasattr(default_manager, 'get_by_natural_key') and hasattr(Model, 'natural_key'):
+                natural_key = Model(**d['fields']).natural_key()[0]
```

Comment:
Natural keys can be tuples of any length, selecting just the first element isn't always valid I think?